core/arch/arm/kernel/elf_load.c

```diff
@@ -161,13 +161,19 @@ static TEE_Result copy_to(struct elf_load_state *state,
 	if (!len)
 		return TEE_SUCCESS;
 
-	if (len > dst_size || (len + dst_offs) > dst_size)
+	/* Check for integer overflow */
+	if ((len + dst_offs) < dst_offs || (len + dst_offs) > dst_size)
 		return TEE_ERROR_SECURITY;
 
+	/* Check for source and destination buffer overflow */
 	if (!core_is_buffer_inside(state->nwdata + offs, len,
 				   state->nwdata, state->nwdata_len))
 		return TEE_ERROR_SECURITY;
 
+	if (!core_is_buffer_inside((uint8_t *)dst + dst_offs, len,
```

Comment:
@jenswi-linaro indeed, the prevous code was ok, without the test added at line 164. Do you think it is ok to simply remove code at line 164, and leave the other changes?